content/blog/working-with-benk-a-storage-provisioning-and-io-performance-benchmark-suite-for-kubernetes.md

```diff
@@ -0,0 +1,232 @@
+  jobs:
+    - threads: 1
+      runtime: 55s
+      iops: 1664
+      bandwidth: 13MB/s
+      bs: 8k
+    - threads: 2
+      runtime: 49s
+      iops: 3291
+      bandwidth: 26MB/s
+      bs: 8k
+    - threads: 4
+      runtime: 48s
+      iops: 8044
+      bandwidth: 63MB/s
+      bs: 8k
+    - threads: 8
+      runtime: 53s
+      iops: 12075
+      bandwidth: 94MB/s
+      bs: 8k
+    - threads: 16
+      runtime: 46s
+      iops: 16357
+      bandwidth: 128MB/s
+      bs: 8k
+    - threads: 32
+      runtime: 51s
+      iops: 17284
+      bandwidth: 135MB/s
+      bs: 8k
+    - threads: 64
+      runtime: 48s
+      iops: 17489
+      bandwidth: 137MB/s
+      bs: 8k
+    - threads: 128
+      runtime: 54s
+      iops: 18761
+      bandwidth: 147MB/s
+      bs: 8k
+```
+
+We can now observe that we’re seeing diminishing returns by adding more than 16 threads to this workload.
+
+# A/B comparison
+
+A/B testing is something that has gained popularity in human interaction testing for websites. This is also a crucial testing methodology to analyze systems performance by simply changing a single parameter and rerun the exact same test to compare the outcomes. We’re in luck as Benk allows reporting on two log files at once with just a slightly different data structures being fed to the Jinja2 templates.
+
+Let’s change block size for our previous workload example. We’re going to summarize if increasing the block size will increase the bandwidth for the workload.
+
+Change the “workloadBlockSize” to “64k” with the `vi` ceremony.
+
+```text
+vi kustomize/overlays/mytest-*/config.env
+```
+
+Re-run the sequencer.
+
+```text
+./sequencer.sh mytest-
+```
+
+We’ll have to use a different template as Jinja2 now has to deal with managing two log files at a time. The syntax for the `outputter.py` script is also slightly different.
+
+```text
+./src/benk/outputter.py -a logs/run-mytest-<unique timestamp>.log -b logs/run-mytest-<unique timestamp>.log -t jinja2/example-default-ab.md.j2
+```
+
+This report template will summarize the findings in a markdown table, suitable to include in a GitHub pull request or similar to illustrate a certain discovery.
+
+```text
+| Threads | A (MB/s) | B (MB/s) | Diff |
+| ------- | -------- | -------- | ---- |
+| 1       | 13       | 55       | 4.2x |
+| 2       | 26       | 106      | 4.1x |
+| 4       | 63       | 181      | 2.9x |
+| 8       | 94       | 450      | 4.8x |
+| 16      | 128      | 661      | 5.2x |
+| 32      | 135      | 748      | 5.5x |
+| 64      | 137      | 840      | 6.1x |
+| 128     | 147      | 833      | 5.7x |
+```
+
+Clearly, the performance increases nearly 5x across the board and we also have a discrepancy in the dataset. Which one?
+
+A good exercise for the reader could be to factor in latency in the report to understand the impact, which there will be as performance plateaus when you add more workload usually result in higher latency. How high?
```

Comment:
You've been addressing the reader as "you" in the blog post. You should keep that going. I'd just change it to:
"A good exercise would be to factor in latency into the report in order to understand the impact, which there will be, as performance plateaus when you add more workloads - usually as a result of higher latency. But, how high?"

(Note: there are a number of changes in the above)